syntax = "proto2";

package fySystemSettingsInfo; //系统设置信息

message  DateTime    //日期
{
	required uint32 date_time = 1;        // 系统时间
}

message DateTimeVerify	//确认设置日期成功
{
	required bool check_date_time = 1;	//状态返回
}

message MilesClearSettings //里程清零
{
	required bool trip_miles_clear = 1;   // 清零小计里程
}

message MilesClearVerify//确认里程清零成功
{
	required bool check_miles_clear = 1;	//状态返回
}

message UpgradeSettings //升级
{
	required int32 msg_id = 1;          // 消息类型
	optional string msg_content = 2;      // 消息内容
	enum MsgType{                         // 类型枚举
	    NORMAL_STATE = 0;
	    NEWEST_VERSION = 1;     // send to mcu, new version found, msg: ui, mcu, ui:mcu
	    UPDATE_NOTIFY = 2;      // recv from mcu, should begin update, msg: NULL
	    UPDATING_STATE = 3;     // send to mcu, update process, msg: ui:90, mcu:40
	    UPDATED_SUCCESS = 4;    // send to mcu, update successfully, msg: NULL
	    UPDATED_FAILED = 5;     // send to mcu, update fail, msg: fail reason, success as "udisk remove".
	} 
}

message CloseBuzzerSettings	//关闭蜂鸣器
{
	required bool close_buzzer = 1;	//手动关闭蜂鸣器
}

message CloseBuzzerVerify	//确认操作蜂鸣器成功
{
	required bool check_close_buzzer = 1; //状态返回
}

message IntensityCtrl	//亮度调节
{
	required uint32 intensity_ctrl = 1;	//设置屏幕亮度
}

message QuenchLCD //关闭屏幕
{
	required uint32 quench_lcd = 1;		//关闭显示屏幕
}

message VideoSwitch	//视频信号切换
{
	required uint32 video_switch = 1;	//0:通道1, 1:通道2, 2:通道3, 3:通道4	
}

